### ISSUE TITLE
Should not warn when using mutable variables

### ISSUE THREAD

**Issue by Pauan** [fable-compiler/Fable#496]
Consider this program:

```
open Fable.Import

do
    let mutable foo = 1
    Browser.console.log (fun _ -> foo)
```

When I compile it in Fable with `"dll": true` I get the following warning:

```
The mutable local 'foo' is implicitly allocated as a reference cell because it has been captured by a closure. This warning is for informational purposes only to indicate where implicit allocations are performed. (L4,16-L4,19) (Main.fs)
```

According to [this site](https://blogs.msdn.microsoft.com/fsharpteam/2014/11/12/announcing-a-preview-of-f-4-0-and-the-visual-f-tools-in-vs-2015/) F# changed the behavior of mutable variables. It used to be that you had to use `ref` for heap-allocated variables, but now the F# compiler auto-detects whether the variable should be stack or heap allocated.

When the F# compiler decides to make the variable heap-allocated, it displays a helpful warning letting you know that there is implicit allocation happening.

That's great for F#, but it's completely useless for Fable, because JavaScript variables are always heap allocated anyways.

The only way to silence the warning is to use a `ref` instead, which is less efficient in JavaScript (because of the extra wrapping/unwrapping).

I suggest having Fable silence this warning. Perhaps it's as simple as adding `--nowarn:3180` ?

**Comment by alfonsogarciacaro** [comment]
I had a similar thing with unused variables. I'm not sure why there're different warnings when generating the dll and parsing the AST (in both cases warning level 3 is used, I think). But in any case, it's probably better to just only display dll warnings in verbose mode. Other _more serious_ warnings will be displayed when reading the AST anyways. This behaviour is introduce in fable-compiler@0.7.5-alpha.15.
